src/main/java/com/amazon/ion/impl/bin/IonManagedWriter_1_1.kt

```diff
@@ -843,23 +852,7 @@ internal class IonManagedWriter_1_1(
         }
     }
 
-    private fun startSystemMacro(macro: SystemMacro) {
-        var macroName = macro.macroName
-        // TODO: Replace `indexOf` with something that is not O(n) time complexity.
-        var id = macroNames.indexOf(macroName)
-        if (id < 0) {
-            // If the name is not in use, put it into the user macro table
-            id = getOrAssignMacroAddressAndName(macroName, macro)
-        }
-        if (macrosById[id] == macro) {
-            // The name and id in the local table refers to the system macro we want to invoke,
-            // so invoke as a local symbol since it's almost always shorter.
-            startMacro(macroName, id, macro)
-        } else {
-            // The name is already being used by something else, so invoke using the system macro syntax.
-            userData.stepInEExp(macro)
-        }
-    }
+    private fun startSystemMacro(macro: SystemMacro) = userData.stepInEExp(macro)
```

Comment:
🗺️ This removes the behavior `when a system macro is invoked, it should be added to the user table if there is no name conflict`, matching test removed below. We can restore this behavior at need, but it's not certain that the user *wants* to add every invoked system macro to their macro table.